src/main/java/com/allianceair/gims/service/ItemService.java

```diff
@@ -1,45 +1,73 @@
 package com.allianceair.gims.service;
 
+import com.allianceair.gims.dto.InventoryDto;
+import com.allianceair.gims.model.Category;
 import com.allianceair.gims.model.InventoryItem;
 import com.allianceair.gims.model.ServiceOrder;
+import com.allianceair.gims.model.Type;
+import com.allianceair.gims.repository.CategoryRepository;
 import com.allianceair.gims.repository.InventoryItemRepository;
+import com.allianceair.gims.repository.TypeRepository;
 import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
+@Slf4j
 @Service
 @AllArgsConstructor
 public class ItemService {
 
     private final InventoryItemRepository inventoryItemRepository;
+    private final CategoryRepository categoryRepository;
+    private final TypeRepository typeRepository;
 
     public InventoryItem addItem(InventoryItem item) {
         return inventoryItemRepository.save(item);
     }
 
-    public List<InventoryItem> getItems() {
-        return inventoryItemRepository.findAll();
+    public List<InventoryDto> getItems() {
+        return mapToDtos(inventoryItemRepository.findAll());
     }
 
-    public List<InventoryItem> getItemsByName(String name) {
-        return inventoryItemRepository.findByNameStartsWith(name);
+    public List<InventoryDto> getItemsByName(String name) {
+        return mapToDtos(inventoryItemRepository.findByNameStartsWithIgnoreCase(name));
     }
-    public List<InventoryItem> getItemsByNameAndCategory(String name, String category) {
-        return inventoryItemRepository.findAllByNameAndCategory(name, category);
+    public List<InventoryDto> getItemsByNameAndCategory(String name, String categoryName) {
+        List<Category> categories = categoryRepository.findByNameStartsWithIgnoreCase(categoryName);
+        List<InventoryItem> items = new ArrayList<>();
+
+        categories.forEach(category -> items.addAll(inventoryItemRepository.findAllByNameAndCategory(name, category.getId())));
+
+        return mapToDtos(items);
     }
 
-    public List<InventoryItem> getItemsByCategory(String category) {
-        return inventoryItemRepository.findByCategoryStartsWith(category);
+    public List<InventoryDto> getItemsByCategory(String categoryName) {
+        List<Category> categories = categoryRepository.findByNameStartsWithIgnoreCase(categoryName);
+        List<InventoryItem> items = new ArrayList<>();
+
+        log.debug("found categories {}", categories);
+
+        //NOTE: ok so I did this part second and I am liking it less and am thinking to store the category name in the inventory table even though it is normalized
```

Comment:
Copying my note in here to make it move visible.

ok so I did this part second and I am liking it less and am thinking to store the category name in the inventory table even though it is normalized

Reply:
I suppose this question depends on whether or not category is theoretically an enum or if it could have metadata associated with it? I'm no longer familiar with the normal forms, so I can't say what I am trying to say in relational terms well. I would think category is just a handful of values and doesn't need to be its own document.